src/basic/linux/update.sh

```diff
@@ -3,9 +3,31 @@
 set -eu
 set -o pipefail
 
+# The directory must be userspace kernel header directory:
+# git clone git@github.com:torvalds/linux.git
+# make -C linux headers
+# ./update.sh linux/usr/include/linux
+SRCDIR=${1?}
+
 for i in *.h */*.h; do
-    curl --fail "https://raw.githubusercontent.com/torvalds/linux/master/include/uapi/linux/$i" -o "$i"
+    if [[ "$i" == bpf_insn.h ]]; then
+        cp "$SRCDIR/samples/bpf/$i" "$i"
+    else
+        cp "$SRCDIR/usr/include/linux/$i" "$i"
+    fi
 
-    sed -r -i -e 's/__user //g' -e '/^#include <linux\/compiler(_types)?.h>/ d' "$i"
-    sed -r -i 's/^(#include <linux\/fs\.h>)/#if WANT_LINUX_FS_H\n\1\n#endif/' "$i"
+    case "$i" in
+        btrfs.h)
+            sed -r -i 's/^(#include <linux\/fs\.h>)/#if WANT_LINUX_FS_H\n\1\n#endif/' "$i"
+            ;;
+        auto_dev-ioctl.h)
+            sed -r -i '/^#define[[:space:]]+AUTOFS_DEV_IOCTL_VERSION_MINOR/ s/[0-9]+$/0/' "$i"
+            ;;
+        ethtool.h)
+            sed -r -i '/return (ep->speed_hi << 16) | ep->speed;/ s/return .*;/return ((__u32) ep->speed_hi << 16) | (__u32) ep->speed;/' "$i"
+            ;;
+        dm-ioctl.h)
+            sed -r -i '/#define[[:space:]]+DM_VERSION_MINOR/ s/[0-9]+/27/' "$i"
```

Comment:
```suggestion
            sed -r -i '/^#define[[:space:]]+DM_VERSION_MINOR/ s/[0-9]+$/27/' "$i"
```

To be consistent with auto_dev-ioctl.h above?